cmake/ExternalLibs.cmake

```diff
@@ -45,7 +45,7 @@ add_library(RapidJSON::rapidjson ALIAS rapidjson)
 
 # Google C++ tests
 if(BUILD_CPP_TEST)
-  find_package(GTest)
+  find_package(GTest REQUIRED CONFIG)
```

Comment:
We shouldn't put `REQUIRED` here, since we want to optionally fetch gtest from GitHub if it's not available in the system prefix.

Reply:
Also, what does `CONFIG` do here? Is it necessary?

Reply:
Thanks! `REQUIRED` was a debugging artifact; I've just removed it. `CONFIG` switches us from [module mode](https://cmake.org/cmake/help/latest/command/find_package.html#basic-signature-and-module-mode) to [config mode](https://cmake.org/cmake/help/latest/command/find_package.html#full-signature-and-config-mode) for package discovery. Config mode is generally the more modern and preferred way to set up packages in CMake, but they still default to module mode to allow library developers to easily override default configs generated by CMake.

In the case of GTest, module mode discovery does not correctly bring in GMock, only GTest. Switching to config mode is the easiest way to properly add GMock to the project.